### ISSUE TITLE
PowerShell command line completion

### ISSUE THREAD

**Issue by stax76** [mpv-player/mpv#12957]
I wrote a PowerShell command line completion script, which can be found here:

https://github.com/stax76/mpv-scripts

**Comment by rubyFeedback** [comment]
Should it be included into mpv? E. g. some subdirectory where it may fit, scripts/ or tools/ or something like that.

**Comment by kasper93** [comment]
Nice, should fix #12334, could you open pull request and put the file in `/etc` directory? I think it is fine to have it in main repo same as zsh and bash completion.

**Comment by stax76** [comment]
Test on Linux and code review revealed issues, so I made a large rewrite today. Pull request is online. Maybe this topic can stay open for a while, which would help users to find it. It could also be added to the changelog.

**Comment by stax76** [comment]
I found out that PowerShell has a menu based autocomplete mode which can be enabled with:

```PowerShell
Set-PSReadlineKeyHandler -Key Tab -Function MenuComplete
```
It's described in an article worthwhile to read:

https://techcommunity.microsoft.com/t5/itops-talk-blog/autocomplete-in-powershell/ba-p/2604524

nushell, which I also use and really like has this enabled by default, I don't know if bash and zsh have that too as an option or extension.

What I also highly recommend is a fzf based history menu, this can be done with most if not all shells and in PowerShell the easiest way is:

```PowerShell
# Install-Module -Name PSFzf
# Set PsFzf's history and file finder keyboard shortcuts.
$env:_PSFZF_FZF_DEFAULT_OPTS = '--exact --no-sort --layout=reverse'
Set-PsFzfOption -PSReadlineChordReverseHistory 'Ctrl+r' -PSReadlineChordProvider 'Ctrl+p'
```

**Comment by Hrxn** [comment]
> I found out that PowerShell has a menu based autocomplete mode which can be enabled with:
> 
> ```powershell
> Set-PSReadlineKeyHandler -Key Tab -Function MenuComplete
> ```

For the record, this is part of PSReadLine, which is included by default in all PS distributions (since x.x, dunno, at least in all 7.x versions), and this feature *is already enabled by default*.

```
PS C:\> Get-PSReadLineKeyHandler -Chord Ctrl+Spacebar

Completion functions
====================

Key           Function     Description
---           --------     -----------
Ctrl+Spacebar MenuComplete Complete the input if there is a single completion, otherwise complete the input by selecting from a menu of possible completions.

PS C:\>
```




> What I also highly recommend is a fzf based history menu, this can be done with most if not all shells and in PowerShell the easiest way is:
> 
> ```powershell
> # Install-Module -Name PSFzf
> # Set PsFzf's history and file finder keyboard shortcuts.
> $env:_PSFZF_FZF_DEFAULT_OPTS = '--exact --no-sort --layout=reverse'
> Set-PsFzfOption -PSReadlineChordReverseHistory 'Ctrl+r' -PSReadlineChordProvider 'Ctrl+p'
> ```

Not here to have an opinion on PSFzf (and fzf), but I don't think that including additional external dependencies is the way to go.
Please keep it to PS default.

**Comment by kasper93** [comment]
Fixed by #12967
